crafting-dead-core/src/main/java/com/craftingdead/core/client/gui/widget/button/CompositeButton.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Crafting Dead
+ * Copyright (C) 2022  NexusNode LTD
+ *
+ * This Non-Commercial Software License Agreement (the "Agreement") is made between
+ * you (the "Licensee") and NEXUSNODE (BRAD HUNTER). (the "Licensor").
+ * By installing or otherwise using Crafting Dead (the "Software"), you agree to be
+ * bound by the terms and conditions of this Agreement as may be revised from time
+ * to time at Licensor's sole discretion.
+ *
+ * If you do not agree to the terms and conditions of this Agreement do not download,
+ * copy, reproduce or otherwise use any of the source code available online at any time.
+ *
+ * https://github.com/nexusnode/crafting-dead/blob/1.18.x/LICENSE.txt
+ *
+ * https://craftingdead.net/terms.php
+ */
+
+package com.craftingdead.core.client.gui.widget.button;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import java.util.function.Function;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.resources.ResourceLocation;
+import org.jetbrains.annotations.NotNull;
+
+public class CompositeButton extends Button {
+
+  private final ResourceLocation textureLocation;
+  private final int atlasX;
+  private final int atlasY;
+  private final int atlasXHover;
+  private final int atlasYHover;
+  private final int atlasXInactive;
+  private final int atlasYInactive;
+
+  private CompositeButton(Builder builder) {
+    super(builder.x, builder.y, builder.width, builder.height, builder.text, builder.action);
+    this.textureLocation = builder.textureLocation;
+    this.atlasX = builder.atlasX;
+    this.atlasY = builder.atlasY;
+    this.atlasXHover = builder.atlasXHover;
+    this.atlasYHover = builder.atlasYHover;
+    this.atlasXInactive = builder.atlasXInactive;
+    this.atlasYInactive = builder.atlasYInactive;
+  }
+
+  public static Builder button(int x, int y, int width, int height,
+      ResourceLocation textureLocation) {
+    return new Builder(x, y, width, height, textureLocation, CompositeButton::new);
+  }
+
+  @Override
+  public void renderButton(@NotNull PoseStack matrixStack, int mouseX, int mouseY,
```

Comment:
Don't need notnulls (unless everything was nullable by default in the package-info)

Reply:
I added it because Intellij idea warns if a parameter that is supposed to be not null has no not null annotation